packages/block-editor/src/components/block-pattern-picker/index.js

```diff
@@ -0,0 +1,195 @@
+	let content;
+	// Render `carousel` in single viewMode.
+	if ( viewMode === 'single' ) {
+		const getSlideClass = ( index ) => {
+			if ( index === activeSlide ) return 'active-slide';
+			if ( index === activeSlide - 1 ) return 'previous-slide';
+			if ( index === activeSlide + 1 ) return 'next-slide';
+			return '';
+		};
+		content = (
+			<div className="block-layout-setup__container">
+				<ul className="carousel-container">
+					{ patterns.map( ( pattern, index ) => {
+						return (
+							<BlockPatternSlide
+								viewMode="single"
+								className={ getSlideClass( index ) }
+								key={ pattern.name }
+								pattern={ pattern }
+								onSelect={ onBlockPatternSelect }
+								composite={ composite }
+							/>
+						);
+					} ) }
+				</ul>
+			</div>
+		);
+	} else {
+		content = (
+			<Composite
+				{ ...composite }
+				role="listbox"
+				className="block-layout-setup__container"
+				aria-label={ __( 'Layout list' ) }
+			>
+				{ patterns.map( ( pattern ) => (
+					<BlockPattern
+						key={ pattern.name }
+						pattern={ pattern }
+						onSelect={ onBlockPatternSelect }
+						composite={ composite }
+					/>
+				) ) }
+			</Composite>
+		);
+	}
+	return content;
+};
+
+function BlockPattern( { pattern, onSelect, composite } ) {
+	// TODO check viewportWidth. From pattern? From resizeObserver to have current width
+	// and manipulate later??
+	const { content } = pattern;
+	const blocks = useMemo( () => parse( content ), [ content ] );
+	const descriptionId = useInstanceId(
+		BlockPattern,
+		'block-setup-block-layout-list__item-description'
+	);
+	return (
+		<div
+			className="block-setup-block-layout-list__list-item"
+			aria-label={ pattern.title }
+			aria-describedby={ pattern.description ? descriptionId : undefined }
+		>
+			<CompositeItem
+				role="option"
+				as="div"
+				{ ...composite }
+				className="block-setup-block-layout-list__item"
+				onClick={ () => onSelect( blocks ) }
+			>
+				<BlockPreview blocks={ blocks } viewportWidth={ 900 } />
+			</CompositeItem>
+			{ !! pattern.description && (
+				<VisuallyHidden id={ descriptionId }>
+					{ pattern.description }
+				</VisuallyHidden>
+			) }
+		</div>
+	);
+}
+
+function BlockPatternSlide( { className, pattern } ) {
+	const { content } = pattern;
+	const blocks = useMemo( () => parse( content ), [ content ] );
+	const descriptionId = useInstanceId(
+		BlockPatternSlide,
+		'block-setup-block-layout-list__item-description'
+	);
+	return (
+		<li
+			className={ `pattern-slide ${ className }` }
+			aria-label={ pattern.title }
+			aria-describedby={ pattern.description ? descriptionId : undefined }
+		>
+			<BlockPreview
+				blocks={ blocks }
+				__experimentalLive
+				// viewportWidth={ viewportWidth }
+			/>
+			{ !! pattern.description && (
+				<VisuallyHidden id={ descriptionId }>
+					{ pattern.description }
+				</VisuallyHidden>
+			) }
+		</li>
+	);
+}
+
+const BlockPatternPicker = ( {
```

Comment:
Given we have the block pattern inserter and soon would need a full-screen pattern browser for the directory, it seems the name should be more clear. If the design for all pattern browsing coalesces a bit more it might be fine. This is more of a design consideration, but let's be sure to make the components clear.

Reply:
I have renamed to `BlockPatternSetup` in: https://github.com/WordPress/gutenberg/pull/29602/commits/4cb03a9e334f568b7dfb125f40150c596c7a7002